content/blog/dbplyr-2-0-0/index.Rmd

```diff
@@ -0,0 +1,204 @@
+
+You can install it from CRAN with:
+
+```{r, eval = FALSE}
+install.packages("dbplyr")
+```
+
+This blog post covers the major improvements in this version:
+
+-   dplyr 1.0.0 compatibility so you can now use `across()`, `relocate()`, `rename_with()`, and more.
+
+-   The major improvements to SQL translation.
+
+-   A snazzy new logo from [Allison Horst](https://www.allisonhorst.com).
+
+-   An improved extension system.
+
+Please see the [release notes](https://github.com/tidyverse/dbplyr/releases/tag/v2.0.0) for a full list of changes.
+
+```{r setup, message = FALSE}
+library(dbplyr)
+library(dplyr, warn.conflicts = FALSE)
+```
+
+## dplyr 1.0.0 compatibility
+
+dbplyr now supports all relevant features added in dplyr 1.0.0:
+
+-   `across()` is now translated into individual SQL statements.
+
+    ```{r}
+    lf <- lazy_frame(g = 1, a = 1, b = 2, c = 3)
+    lf %>% 
+      group_by(g) %>% 
+      summarise(across(everything(), mean, na.rm = TRUE))
+    ```
+
+-   `rename()` and `select()` support dplyr 1.0.0 tidyselect syntax, apart from predicate functions which can't easily work on computed queries.
+    You can now use `rename_with()` to programmatically rename columns.
+
+    ```{r}
+    lf <- lazy_frame(x1 = 1, x2 = 2, x3 = 3, y1 = 4, y2 = 3)
+    lf %>% select(starts_with("x") & !"x3")
+    lf %>% select(ends_with("2") | ends_with("3"))
+    lf %>% rename_with(toupper)
+    ```
+
+-   `relocate()` makes it easy to move columns around:
+
+    ```{r}
+    lf <- lazy_frame(x1 = 1, x2 = 2, y1 = 4, y2 = 3)
+    lf %>% relocate(starts_with("y"))
+    ```
+
+-   `slice_min()`, `slice_max()`, and `slice_sample()` are now supported, and `slice_head()` and `slice_tail()` throw informative error messages (since they don't make sense for databases).
+
+    ```{r}
+    lf <- lazy_frame(g = rep(1:2, 5), x = 1:10)
+    lf %>% 
+      group_by(g) %>% 
+      slice_min(x, prop = 0.5)
+
+    lf %>% 
+      group_by(g) %>% 
+      slice_sample(x, n = 10, with_ties = TRUE)
+    ```
+
+    Note that these slices are translated to window functions, and because you can't use a window function directly inside a `WHERE` clause, they must be wrapped in a subquery.
+
+## SQL translation
+
+The dbplyr documentation now does a much better job of providing the details of its SQL translation.
+Each backend and each major verb has a documentation page giving the basics of the translation.
+This will hopefully make it much easier to learn what is and isn't supported by dbplyr.
+Visit <https://dbplyr.tidyverse.org/reference/index.html> to see the new docs.
+
+There were also many improvements to SQL generation.
+Here are a few of the most important:
+
+-   Join functions gain a `na_matches` argument that allows you to control whether or not `NA` (`NULL`) values match other `NA` values.
```

Comment:
```suggestion
-   Join functions gain an `na_matches` argument that allows you to control whether or not `NA` (`NULL`) values match other `NA` values.
```

Reply:
I mentally read `na_matches` as _N-A matches_, so I'm giving it an "an".

Reply:
Yeah, agreed